src/mongoc/mongoc-uri.c

```diff
@@ -42,7 +42,8 @@ struct _mongoc_uri_t
    char                   *database;
    bson_t                  options;
    bson_t                  credentials;
-   bson_t                  read_prefs;
+   bson_t                  read_prefs_tags;
```

Comment:
We can certainly fetch the final tag set from the `mongoc_read_prefs_t` struct, but `mongoc_uri_parse_tags()` is invoked each time `readPreferenceTags` is encountered during query string parsing and needs a `bson_t` to build upon.

If we initialize the `mongoc_read_prefs_t` struct earlier (i.e. during URI initialization), instead of in `_mongoc_uri_build_read_prefs()`, we can have `mongoc_uri_parse_tags()` call `mongoc_read_prefs_add_tag()` each time.

The actual mode should still be set just once in `_mongoc_uri_build_read_prefs()`, after all query string parameters are parsed, so that we can enforce `readPreference` taking priority over `slaveOk`.